src/System.Device.Gpio/System/Device/Gpio/Drivers/LibgpiodCapiDriver.cs

```diff
@@ -0,0 +1,269 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Runtime.InteropServices;
+
+namespace System.Device.Gpio.Drivers
+{
+    public class LibgpiodCapiDriver : GpioDriver
+    {
+        private const string BasePath = "/dev/";
+
+        public string GpioChip { get; set; } = "gpiochip0";
+
+        private SafeChipHandle _chip;
+
+        private Dictionary<int, SafeLineHandle> _pinNumberToSafeLineHandle;
+
+        private timespec _defaultTimeOut = new timespec(IntPtr.Zero, new IntPtr(1_000_000)); // one millisecond
+
+        private Dictionary<int, LibgpiodDriverEventHandler> _pinNumberToEventHandler = new Dictionary<int, LibgpiodDriverEventHandler>();
+
+        protected internal override int PinCount
+        {
+            get
+            {
+                return Interop.gpiod_chip_num_lines(_chip);
+            }
+        }
+
+        public LibgpiodCapiDriver()
+        {
+            OpenChip();
+            _pinNumberToSafeLineHandle = new Dictionary<int, SafeLineHandle>(PinCount);
+        }
+
+        public LibgpiodCapiDriver(string chip)
+        {
+            if (chip != null)
+                GpioChip = chip;
+            OpenChip();
+            _pinNumberToSafeLineHandle = new Dictionary<int, SafeLineHandle>(PinCount);
+        }
+
+        private void OpenChip()
+        {
+            _chip = Interop.gpiod_chip_open($"{BasePath}{GpioChip}");
+            if (_chip == null)
+            {
+                throw new GpioChipException($"Chip: {GpioChip} not available, error code: {Marshal.GetLastWin32Error()}");
+            }
+        }
+
+        private void RequestEvent(int pinNumber, LibgpiodDriverEventHandler eventDescriptor, PinEventTypes eventType)
+        {
+            int eventSuccess = -1;
+            SafeLineHandle pin = Interop.gpiod_chip_get_line(_chip, pinNumber);
+            if (pin != null)
+            {
+                eventDescriptor.PinHandle = pin;
+                if (eventType == PinEventTypes.Rising || eventType == PinEventTypes.Falling)
+                    eventSuccess = Interop.gpiod_line_request_both_edges_events(pin, $"Listen {pinNumber} for falling edge event");
+                else
+                    throw new GpioChipException("Invalid or not supported event type requested");
```

Comment:
Although our coding guidelines aren't really explicit about it, in corefx we generally only omit braces on conditional blocks where there's not only one line, but there's also no else block. So this
```c#
if (foo)
    Baz();
```
but not
```c#
if (foo)
    Baz();
else
    Boz();
```

Un-braced conditional blocks are dangerous for the developer maintaining the code because it is easy to add another line and assume it's part of the block, ie.,
```c#
if (foo)
    Baz();
else
    Boz();
    Oops();
```
Arguably it's easier to spot this when there's only an `if` and no `else`. Original discussion [here](https://github.com/dotnet/corefx/issues/381) It's not explicitly stated as a rule though so up to you.

Reply:
got it, thanks